crates/bpe/src/byte_pair_encoding.rs

```diff
@@ -541,19 +554,73 @@ mod tests {
             }
         }
     }
+}
+
+#[cfg(test)]
+mod data {
+    use std::collections::HashSet;
+    use std::fs::File;
+    use std::path::PathBuf;
+
+    use rand::Rng;
+    use serde::Serialize;
+    use tiktoken_rs::{cl100k_base, o200k_base};
+
+    use super::*;
+
+    const BPE_CL100K_LEN: usize = 100256;
+    const BPE_O200K_LEN: usize = 199998;
+
+    /// Use this to find a hashing factor for [`hash_bytes`] that prevents collisions.
+    /// 1. Ensure all supported tokenizers are in the list.
+    /// 2. Update the hash factor in [`hash_bytes`].
+    /// 3. Run [`update_token_dicts`] tests below to update data files.
+    #[test]
+    #[ignore = "run manually to find a suitable hash factor"]
+    fn find_hash_factor() {
+        let bpes: &mut [(CoreBPE, usize)] = &mut [
+            (cl100k_base().unwrap(), BPE_CL100K_LEN),
+            (o200k_base().unwrap(), BPE_O200K_LEN),
+        ];
+        let mut rnd = rand::thread_rng();
+        loop {
+            let factor: u64 = rnd.gen();
+            if bpes.iter().all(|(bpe, len)| {
+                let mut seen = HashSet::with_capacity(*len);
+                (0..*len)
+                    .all(|i| seen.insert(hash_bytes_with_factor(&bpe._decode_native(&[i]), factor)))
+            }) {
+                println!("hash factor: {factor}");
+                return;
+            }
+        }
+    }
 
-    // TODO: Move the generation of the dictionary into some build procedure?
     #[test]
-    fn test_serialize() {
+    #[ignore = "run manually to update data files"]
```

Comment:
in principle, we could let this test run normally, since it will fix the broken data file (and one will see in the diff that something has changed)

Reply:
Yes, let's do that.